### ISSUE TITLE
PersistentVolumeClaimExample : don't work

### ISSUE THREAD

**Issue by survivant** [fabric8io/kubernetes-client#2071]
there is few errors in PersistentVolumeClaimExample.  

line 81 and 83 you have that this : 

```
client.persistentVolumes().create(pv);

client.persistentVolumes().create(pv);
```

it will failed because the pv already exists.

and in the finally block

you should have this : 

```
finally {
        client.persistentVolumeClaims().withName("test-pv-claim").delete();
        client.persistentVolumes().withName("test-local-pv").delete(); //fixed the name
        client.pods().withName("test-pv-pod").delete(); // you forgot to remove the pod
        client.storage().storageClasses().withName("my-local-storage").delete(); // forgot to remove it
      }
```

instead of : 
```
finally {
        client.persistentVolumeClaims().withName("test-pv-claim").delete();
        client.persistentVolumes().withName("test-pv").delete(); // don't exist in this example
      }
```

**Comment by vasudevcoder** [comment]
I want to contribute in this issue. But I don't know how to start. Would I have to make above mentioned changes or something else.

**Comment by vasudevcoder** [comment]
I tried to build the file with above mentioned changes it is Building successfully. Also i runned this command
mvn clean verify -Pitests-kubernetes
This also running successfully. What else I have to do in this issue

**Comment by rohanKanojia** [comment]
@vasudevcoder : You need to check whether this file is running properly or not: 
https://github.com/fabric8io/kubernetes-client/blob/555c3790b02b910d62f72e95350dd563c97359a8/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/PersistentVolumeClaimExample.java#L20-L70

**Comment by vasudevcoder** [comment]
when i tried to run the file it is showing me this error.
Error:java: java.lang.OutOfMemoryError: Java heap space
How can i remove this?

**Comment by rohanKanojia** [comment]
@vasudevcoder : You need to increase your IDE's heap size. For IntelliJ, it can be done like [this](https://stackoverflow.com/questions/17221725/how-to-increase-the-memory-heap-size-on-intellij-idea)

**Comment by vasudevcoder** [comment]
> @vasudevcoder : You need to increase your IDE's heap size. For IntelliJ, it can be done like [this](https://stackoverflow.com/questions/17221725/how-to-increase-the-memory-heap-size-on-intellij-idea)

i tried this but it is showing me same error

**Comment by rohanKanojia** [comment]
How much heap size have you allocated for your IDE? In my case it's 8g
![Screenshot from 2020-04-05 12-49-44](https://user-images.githubusercontent.com/13834498/78469137-f4ad5f00-773b-11ea-887b-6590f0eb6601.png)

**Comment by vasudevcoder** [comment]
> How much heap size have you allocated for your IDE? In my case it's 8g
> ![Screenshot from 2020-04-05 12-49-44](https://user-images.githubusercontent.com/13834498/78469137-f4ad5f00-773b-11ea-887b-6590f0eb6601.png)

I allocated only 700 MB.

**Comment by vasudevcoder** [comment]
It is now running without any error but it showing some warnings.
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Using master with url : https://localhost:8443/
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Could not create resource: Operation: [delete]  for kind: [PersistentVolumeClaim]  with name: [test-pv-claim]  in namespace: [null]  failed.

Should i ignore them.

**Comment by vasudevcoder** [comment]
I tried building file it is building successfully also this command mvn clean verify -Pitests-kubernetes also running successfully.

**Comment by vasudevcoder** [comment]
> @vasudevcoder : You need to check whether this file is running properly or not:

Yes it is running properly

**Comment by rohanKanojia** [comment]
`mvn clean verify -Pitests-kubernetes` runs integration tests. Why are you running it? bdw, this error message shows that maybe something is not right( Read OP's first comment):
```
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Could not create resource: Operation: [delete] for kind: [PersistentVolumeClaim] with name: [test-pv-claim] in namespace: [null] failed.
```
looks like code is falling into this exception block: https://github.com/fabric8io/kubernetes-client/blob/555c3790b02b910d62f72e95350dd563c97359a8/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/PersistentVolumeClaimExample.java#L110 
Could you please add `e.printStackTrace()` there and check what's the exception?

**Comment by vasudevcoder** [comment]
> Could you please add `e.printStackTrace()` there and check what's the exception?


This is showing me this

OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Using master with url : https://localhost:8443/
io.fabric8.kubernetes.client.KubernetesClientException: Operation: [delete]  for kind: [PersistentVolumeClaim]  with name: [test-pv-claim]  in namespace: [null]  failed.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64)
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:72)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.deleteThis(BaseOperation.java:725)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.delete(BaseOperation.java:645)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.delete(BaseOperation.java:70)
	at io.fabric8.kubernetes.examples.PersistentVolumeClaimExample.main(PersistentVolumeClaimExample.java:106)
Caused by: java.net.ConnectException: Failed to connect to localhost/127.0.0.1:8443
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:249)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.BackwardsCompatibilityInterceptor.intercept(BackwardsCompatibilityInterceptor.java:134)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.ImpersonatorInterceptor.intercept(ImpersonatorInterceptor.java:68)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.lambda$createHttpClient$3(HttpClientUtils.java:114)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall.execute(RealCall.java:93)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:469)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:430)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleDelete(OperationSupport.java:231)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.deleteThis(BaseOperation.java:722)
	... 3 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:129)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
	... 33 more
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Could not create resource: Operation: [delete]  for kind: [PersistentVolumeClaim]  with name: [test-pv-claim]  in namespace: [null]  failed.

**Comment by kunal-kushwaha** [comment]
I think this issue is same as #2070

**Comment by kunal-kushwaha** [comment]
I ran the same example and added .withNamespace("default") in the metadata and it worked.

**Comment by vasudevcoder** [comment]
> I ran the same example and added .withNamespace("default") in the metadata and it worked.

Okay I tried to do that.

**Comment by kunal-kushwaha** [comment]
@vasudevcoder : Make sure your master url is correct, are you able to build other resources with the client or having issues with pic only?

**Comment by vasudevcoder** [comment]
Yes I am able to build other resources

**Comment by vasudevcoder** [comment]
> @vasudevcoder : Make sure your master url is correct, are you able to build other resources with the client or having issues with pic only?

My master url is https://localhost:8443/ is it correct or not . And i think i am not able to build other resources too

**Comment by kunal-kushwaha** [comment]
@vasudevcoder : Try running $ kubectl config view

**Comment by vasudevcoder** [comment]
> @vasudevcoder : Try running $ kubectl config view

ok I tried to do that

**Comment by vasudevcoder** [comment]
> @vasudevcoder : Try running $ kubectl config view

`apiVersion: v1
clusters:
- cluster:
    certificate-authority: /home/vasudev/.minikube/ca.crt
    server: https://192.168.99.100:8443
  name: minikube
contexts:
- context:
    cluster: minikube
    user: minikube
  name: minikube
current-context: minikube
kind: Config
preferences: {}
users:
- name: minikube
  user:
    client-certificate: /home/vasudev/.minikube/profiles/minikube/client.crt
    client-key: /home/vasudev/.minikube/profiles/minikube/client.key`

**Comment by kunal-kushwaha** [comment]
@vasudevcoder : Try using 'https://192.168.99.100:8443' as your url in the example.

**Comment by vasudevcoder** [comment]
> @vasudevcoder : Try using 'https://192.168.99.100:8443' as your url in the example.

This is also showing same error

**Comment by vasudevcoder** [comment]
I am able to remove all the errors.
`OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Using master with url : https://192.168.99.100:8443/
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Creating PersistentVolume object
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Successfully created PersistentVolume object
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Creating PersistentVolumeClaim object
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Successfully created PersistentVolumeClaim object
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Creating pod
[main] INFO io.fabric8.kubernetes.examples.PersistentVolumeClaimExample - Successfully created pod`

Also gettting pvc now

`NAME            STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS       AGE
test-pv-claim   Pending                                      my-local-storage   3m52s
`

**Comment by kunal-kushwaha** [comment]
@vasudevcoder: That's great, can you please make a PR?

**Comment by vasudevcoder** [comment]
I already done that

**Comment by kunal-kushwaha** [comment]
@vasudevcoder : You can also put "Fixes #2071" in the comment section of your PR to link it with this issue.
